### ISSUE TITLE
Switching from PHP or HTML block to theme selector leaves the old 'body'  values in

### ISSUE THREAD

**Issue by Bopske** [Tinyportal/TinyPortal#955]
Whe you have a PHP block or a HTML block and switch the block type to a Theme block the block seems ' broken' (see images). As far as I can see the old 'body' values are left in the block. 

We should not only reset the parameter values but also the body value.
Same happens when switching from PHP or HTML to a shoutbox, or (and this is a fun one) if you change a PHP block to a Online block, and THEN change it to a theme selector.

Basically: when changing the block type, the body value should be wiped IMO...

![2023-01-23_214109](https://user-images.githubusercontent.com/24902644/214147297-fe95199e-20a7-460c-a6c1-43bd6a750dec.png)
![2023-01-23_214801](https://user-images.githubusercontent.com/24902644/214147301-041bcda0-c78a-4786-83ea-1e1e0e337630.png)

**Comment by Bopske** [comment]
Had a look at where you are setting the block defaults, but those only apply to the parameters....

**Comment by Bopske** [comment]
@tinoest can you check this?
